virttest/utils_logfile.py

```diff
@@ -150,13 +155,18 @@ def close_log_file(filename):
         raise LogLockError("Could not acquire exclusive lock to access"
                            " _open_log_files")
     try:
-        for k in _open_log_files:
-            if os.path.basename(k) == filename:
-                f = _open_log_files[k]
-                f.close()
-                remove.append(k)
+        for log_file, log_fd in _open_log_files.items():
+            if os.path.basename(log_file) == os.path.basename(filename):
+                log_fd.close()
+                remove.append(log_file)
         if remove:
             for key_to_remove in remove:
                 _open_log_files.pop(key_to_remove)
+
     finally:
         _log_lock.release()
+
+
+def close_own_log_file(self):
+    """Closing hook for sessions whose log_file attribute should be passed along."""
+    close_log_file(self.log_file)
```

Comment:
Then, if we are going to let VT manage these log handlers, I'd suggest introducing a closure to retrieve them instead of accessing with session's attribute. However, firstly we still need to confirm if the close hook could be customized for aexpect objects.

```python
def close_own_log_file(log_file):

    def hook(self):
        close_log_file(log_file)

    return hook
```

Reply:
Indeed, this is a good extra step to get rid of the stale log file attribute of the sessions which could then later be dropped on the aexpect side in favor of the single output function. Pushed with this change in a separate commit that explains the idea and that we will have to test next.